app/routes.py

```diff
@@ -1,2 +1,239 @@
-from flask import Blueprint
+from flask import Blueprint, json, jsonify, request, make_response
+from werkzeug.wrappers import response
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+import requests
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["POST"])
+def post_tasks():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return make_response({"details":"Invalid data"}, 400)
+    
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"],
+        completed_at = request_body["completed_at"]
+        )
+    db.session.add(new_task)
+    db.session.commit()
+    
+    response_body = {"task": {"id": new_task.id,
+        "title": new_task.title,
+        "description": new_task.description,
+        "is_complete": bool(new_task.completed_at)}}
+        
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])   
+def get_tasks():
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks: 
+            tasks_response.append({
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)
+                })
+
+        return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    if not task:
+        return make_response(f"Task {task_id} not found", 404)
+        
+    if request.method == "GET":
+        if not task.goal_id:
+            response_body = {"task":
+                {"id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)}}
+            return jsonify(response_body), 200
+        
+        else:
+            response_body = {"task":
+            {"id": task.id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)}}
+            return jsonify(response_body), 200
```

Comment:
It's a little strange that the tests expected `goal_id` to appear only when it's set. I would typically expect a public API that had relationships like this to simply return `null` for a task that wasn't associated with a `Goal`, but this does the job.

However, we can also notice that including this key is the only difference between the two cases, so rather than duplicating the dictionary code, we could generate the base dictionary, then only add `goal_id` if the value is defined in the `Task`.

Also, should the route be the one that decides which model rendering function to use by inspecting the goal relationship within the task itself? Or should the task simply be asked to convert itself to a dictionary, and then it decides how to build the dictionary given its own internal state? It's not clear either way, but it's definitely worth thinking about!